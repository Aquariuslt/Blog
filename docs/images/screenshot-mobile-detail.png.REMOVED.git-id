24e7e81bb9c56fe04bc2576f74419318e69ef06a