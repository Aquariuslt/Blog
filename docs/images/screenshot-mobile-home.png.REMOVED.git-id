32f3f91b39a3e2a58d2d46668b3f46f3ad18f625