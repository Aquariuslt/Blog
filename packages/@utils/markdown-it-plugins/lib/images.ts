import * as _ from 'lodash';

function detectImages(md): void {
  md.core.ruler.push('detect_images', (state): void => {
    const tokens = state.tokens;
    const images: string[] = [];

    tokens.map((token): void => {
      if (token.type === 'inline') {
        if (token.children) {
          token.children.map((childToken): void => {
            if (childToken.type === 'image') {
              childToken.attrs.map((imageAttr): void => {
                if (_.isArray(imageAttr) && imageAttr.length > 1 && imageAttr[0] === 'src') {
                  const imageUrl = imageAttr[1];
                  images.push(imageUrl);
                }
              });
            }
          });
        }
      }
    });

    if (state.env) {
      state.env.images = images;
    }
  });
}

export default detectImages;
