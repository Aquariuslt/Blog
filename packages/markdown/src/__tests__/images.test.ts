import * as MarkdownIt from 'markdown-it';

import { ImagesDetectionPlugin } from '../images';
import { read } from '../__tests__/test.fixtures.helper';

describe('markdown-it plugins: images', (): void => {
  it('# should detect images at env', (): void => {
    const md = MarkdownIt().use(ImagesDetectionPlugin);
    const raw = read(`sample-article.md`);
    const context = {};
    md.parse(raw, context);

    expect(context).toHaveProperty('images');
    expect(context['images'].length).toBe(2);
    expect(context['images']).toMatchSnapshot();
  });

  it('# should detect no images when no env', (): void => {
    const md = MarkdownIt().use(ImagesDetectionPlugin);
    const raw = read(`sample-article.md`);
    md.parse(raw, undefined);
  });
});
