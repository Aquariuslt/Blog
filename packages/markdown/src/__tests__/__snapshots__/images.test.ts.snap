// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@utils/markdown-it-images # should detect images at env 1`] = `
Array [
  "https://img.aquariuslt.com/posts/2019/08/migrating-github-actions.png",
  "images/sample-image.png",
]
`;
